src/H5M.c

```diff
@@ -461,6 +612,61 @@ H5Mclose(hid_t map_id)
     FUNC_LEAVE_API(ret_value)
 } /* end H5Mclose() */
 
+/*-------------------------------------------------------------------------
+ * Function:    H5Mclose_async
+ *
+ * Purpose:     Asynchronous version of H5Mclose
+ *
+ * Return:      SUCCEED/FAIL
+ *
+ *-------------------------------------------------------------------------
+ */
+herr_t
+H5Mclose_async(const char *app_file, const char *app_func, unsigned app_line, hid_t map_id, hid_t es_id)
+{
+    void *         token     = NULL;    /* Request token for async operation            */
+    void **token_ptr = H5_REQUEST_NULL; /* Pointer to request token for async operation */
+    H5VL_object_t *vol_obj   = NULL;    /* VOL object of dset_id */
+    H5VL_t *       connector = NULL;    /* VOL connector */
+    herr_t ret_value = SUCCEED; /* Return value */
+
+    FUNC_ENTER_API(FAIL)
+    H5TRACE5("e", "*s*sIuii", app_file, app_func, app_line, map_id, es_id);
+
+    /* Check args */
+    if (H5I_MAP != H5I_get_type(map_id))
+        HGOTO_ERROR(H5E_ARGS, H5E_BADTYPE, FAIL, "not a map ID")
+
+    /* Get dataset object's connector */
+    if (NULL == (vol_obj = H5VL_vol_object(dset_id)))
+        HGOTO_ERROR(H5E_MAP, H5E_CANTGET, FAIL, "can't get VOL object for dataset")
+
+    /* Prepare for possible asynchronous operation */
+    if (H5ES_NONE != es_id) {
+        /* Increase connector's refcount, so it doesn't get closed if closing
+         * the dataset closes the file */
+        connector = vol_obj->connector;
+        H5VL_conn_inc_rc(connector);
+
+        /* Point at token for operation to set up */
+        token_ptr = &token;
+    } /* end if */
+
+    /* Decrement the counter on the map.  It will be freed if the count
+     * reaches zero.
+     */
+    if (H5I_dec_app_ref_always_close_async(map_id, token_ptr) < 0)
+        HGOTO_ERROR(H5E_MAP, H5E_CANTDEC, FAIL, "can't decrement count on dataset ID")
+
+    /* If a token was created, add the token to the event set */
+    if (NULL != token)
+        if (H5ES_insert(es_id, vol_obj->connector, token, H5ARG_TRACE5(FUNC, "*s*sIuii", app_file, app_func, app_line, map_id, es_id)) < 0)
+            HGOTO_ERROR(H5E_MAP, H5E_CANTINSERT, FAIL, "can't insert token into event set")
+
+done:
```

Comment:
Need the "decrement the connector refcount" codeblock here.  (See e.g. H5Dclose_async)

Reply:
I have no idea what happened.  For every file, a couple of them were missing.  I'm very sure, I checked and double checked...  I added all now, and am compiling.  I'll commit them soon.  Thank you!

Reply:
Sorry, I wasn't specific enough with my last comment - the H5I_dec_ref_always_close() that you just put in shouldn't be there, but this routine needs the:
   if (connector && H5VL_conn_dec_rc(connector) < 0)
         HDONE_ERROR(H5E_DATATYPE, H5E_CANTDEC, FAIL, "can't decrement ref count on connector")
block right after the 'done:' label.